src/test/testing_assert.cpp

```diff
@@ -33,7 +47,7 @@ Matrix _table_to_matrix(const std::shared_ptr<const opossum::Table>& t) {
       const auto column = chunk.get_column(col_id);
 
       for (opossum::ChunkOffset chunk_offset = 0; chunk_offset < chunk.size(); ++chunk_offset) {
```

Comment:
**General question**: Currently, we have no guidelines about how to initialize variables, especially in for-loops. IN this file there are at least two:

```c++
opossum::ChunkOffset chunk_offset = 0
opossum::ColumnID col_id{0}
// not entirely different but violates our coding guidelines
unsigned row = 0  // uint32_t / size_t should be used
```

I prefer using auto …
```C++
auto chunk_offset = ChunkOffset{0}`
auto row_id = size_t{0}
```

but would also be okay with 
```C++
ChunkOffset chunk_offset{0}`
size_t row_id{0}
```

One advantage of using `auto` is that it renders unintialized variables impossible.

@mrks, when do we use `auto` and when not?

Reply:
I don't think we have a rule for that. Since it does not really change anything outside of that particular line, I don't think we need one. More rules make it easier to forget the more important ones.

Reply:
Alright. In any case, `unsigned row = 0` should be changed.

Reply:
Thanks for the input, I changed it (and some other occurences) to using `auto` now.